compiler/src/dotty/tools/dotc/typer/RefChecks.scala

```diff
@@ -580,6 +580,8 @@ object RefChecks {
         overrideError("may not override non-experimental member")
       else if other.hasAnnotation(defn.DeprecatedOverridingAnnot) then
         overrideDeprecation("", member, other, "removed or renamed")
+      else if !member.hasAnnotation(defn.PublicInBinaryAnnot) && other.hasAnnotation(defn.PublicInBinaryAnnot) then
+        overrideError("also needs to be declared with @publicInBinary")
```

Comment:
This must come before the `overrideDeprecation` test, otherwise a deprecation *warning* can shadow an actual *error*.